pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -191,186 +160,113 @@ func (c *HPAWatcherClient) Start() {
 			}
-			cmList = append(cmList, cm)
 		}
 	}
-	return cmList
-}
-
-// validate queries Datadog to validate the availability of a metric
-func (hpa *HPAWatcherClient) validate(cm CustomExternalMetric) (value int64, valid bool, err error) {
-	val, err := hpa.queryDatadogExternal(cm.Name, cm.Labels)
-	if err != nil {
-		return cm.Value, false, err
-	}
-	return val, true, nil
-}
-
-// storeHPA processes the data collected from the HPA object watch to be validated and stored in the datadogHPAConfigMap ConfigMap.
-func (c *HPAWatcherClient) storeHPA(hpaList []*v2beta1.HorizontalPodAutoscaler) error {
-	listCustomMetrics := c.processHPA(hpaList)
-	cm, err := c.clientSet.CoreV1().ConfigMaps(c.ns).Get(datadogHPAConfigMap, metav1.GetOptions{})
-	if err != nil {
-		log.Errorf("Could not store the custom metrics data in the ConfigMap %s: %s", datadogHPAConfigMap, err.Error())
+	if err = c.store.Update(); err != nil {
+		log.Infof("Could not update the external metrics in the store: %s", err.Error())
 		return err
 	}
-	for _, n := range listCustomMetrics {
-		toStore, _ := json.Marshal(n)
-		if cm.Data == nil {
-			// Don't panic "assignment to entry in nil map" at init
-			cm.Data = make(map[string]string)
-		}
-		// We use a specific key to avoid conflicting when processing several HPA manifests with the same name but in different namespaces.
-		key := fmt.Sprintf("external.metrics.%s.%s-%s", n.HPANamespace, n.HPAName, n.Name)
-		cm.Data[key] = string(toStore)
-	}
+	return nil
+}
 
-	_, err = c.clientSet.CoreV1().ConfigMaps(c.ns).Update(cm)
+// validateExternalMetric queries Datadog to validate the availability of an external metric
+func (c *HPAWatcherClient) validateExternalMetric(em custommetrics.ExternalMetricValue) (value int64, valid bool, err error) {
+	val, err := c.queryDatadogExternal(em.MetricName, em.Labels)
 	if err != nil {
-		log.Infof("Could not update the ConfigMap: %s", err)
+		return em.Value, false, err
 	}
-	return err
+	return val, true, nil
 }
 
-// removeEntryFromConfigMap will remove an External Custom Metric from removeEntryFromConfigMap if the corresponding HPA manifest is deleted.
-func (c *HPAWatcherClient) removeEntryFromConfigMap(deleted []*v2beta1.HorizontalPodAutoscaler) error {
-	cm, err := c.clientSet.CoreV1().ConfigMaps(c.ns).Get(datadogHPAConfigMap, metav1.GetOptions{})
-	if err != nil {
-		log.Errorf("Could not remove the custom metrics data in the ConfigMap %s: %s", datadogHPAConfigMap, err.Error())
-		return err
-	}
-	for _, d := range deleted {
-		for _, m := range d.Spec.Metrics {
-			metricName := m.External.MetricName
-			key := fmt.Sprintf("external.metrics.%s.%s-%s", d.Namespace, d.Name, metricName)
-			if cm.Data[key] != "" {
-				delete(cm.Data, key)
-				log.Debugf("Removed entry %#v from the ConfigMap %s", key, datadogHPAConfigMap)
+// removeEntryFromStore will remove an External Custom Metric from removeEntryFromStore if the corresponding HPA manifest is deleted.
```

Comment:
removeEntryFromStore will remove an External Custom Metric from the local store & the CM ?